app/scripts/modules/core/src/pipeline/config/stages/stage.module.js

```diff
@@ -52,8 +53,12 @@ module.exports = angular.module('spinnaker.core.pipeline.config.stage', [
       selectedStageType: null,
     };
 
-    accountService.listProviders($scope.application).then((providers) => {
-      $scope.options.stageTypes = pipelineConfig.getConfigurableStageTypes(providers);
+    accountService.providerAccounts($scope.application).then(providerAccounts => {
+      const providers = Object.keys(providerAccounts);
+      const accounts = providers.reduce((memo, provider) => memo.concat(providerAccounts[provider]), []);
+      let stageTypes = pipelineConfig.getConfigurableStageTypes(providers);
```

Comment:
My slight preference would be that `getConfigurableStageTypes` returned the exact set of `stageTypes` to render (i.e., this logic should be encapsulated in that method, probably by passing it a list `{ provider: string, providerVersion: string}[]`).

Reply:
:+1: done, thanks, def the better approach.